### DISCUSSION TITLE
Erratic starting of function and unable to stop it once it does

### DISCUSSION THREAD

**Discussion by EmperorKabir** [pybricks/support#1423] [Q&A]
Hi, so I've combined 2 big chunks of code, one was for 2 city hubs, one broadcasting, one receiving, each with a train motor and the broadcasting one with a colour sensor as well which accelerates and decelerates progressively based on coloured tiles.

The other was for a remote control for 2 city hubs each with a train motor, where:
-either red button stopped the train
-left plus set a constant power of 50% while held
-left minus set a constant power of minus 50% while held
-right plus would set power to increase by 10%
-right minus would set power to decrease by 10%

Having put them together, the only other change I've made in terms of function is to set the left red button to run the first chunk of code rather than just be a second killswitch.

The problem:
1: Very rarely does the left red button actuallly begin the relevant bit of code that I want it to; it seems to work only once in a while, perhaps my code is starting the stopping then immediately stopping code as I've already pressed the button and it might be seeing it more than once?
2: Once the program DOES actually start, it cannot be stopped. Pressing any button will not stop it running and it stays stuck in the acceleration, deceleration, stop at station loop which is towards the bottom of the code:

Sending hub code:

![image](https://github.com/pybricks/support/assets/114411808/77c4b4c0-4a49-4469-94df-f3c2b859bb4e)

I also tried a slight variant but I suffer the same issues, (in fact this seems to add another issue of no longer seeing the deceleration tile but that might be because it's going too fast).

![image](https://github.com/pybricks/support/assets/114411808/c5849c0c-81a6-4d4b-b41a-ae3a73d0b4bb)

**Comment by laurensvalk** [comment]
> Very rarely does the left red button actuallly begin the relevant bit of code that I want it to

When it begins running the blocks for the left button, the left button is pressed. This means some (any) button is pressed, so the repeat loop ends right away, and nothing appears to happen. (I think you hypothesized this correctly).

![image](https://github.com/pybricks/support/assets/12326241/05ed1506-2634-43f0-bb3e-a3c8386e1d68)

**Reply by EmperorKabir** [reply]
Aha, thanks, I'm very close to getting my 'final' remote control train code working! I have replaced it with a bunch of 'or' conditions but I was wondering if there is a way of doing 'remote any button except left' rather than all these 'or' blocks. I tried using the 'not' block but I cannot then map a button type to it.

I also have a strange issue where if for example I press right plus and then the left button, the operation for right plus works fine then upon pressing left, it accelerates for a fraction of a second to what seems like full power, stops dead for about 5 seconds, then starts the operation for the left button.

This is my code nowl (it's huge! I'm so proud!)

![image](https://github.com/pybricks/support/assets/114411808/514326cb-2992-420a-9dff-ef96a47332bf)
